src/Controller/RedirectController.php

```diff
@@ -0,0 +1,115 @@
+        LocationService $locationService,
+        SearchService $searchService,
+        ContainerInterface $container,
+        $configResolver
+    ) {
+        $this->contentService = $contentService;
+        $this->contentTypeService = $contentTypeService;
+        $this->locationService = $locationService;
+        $this->searchService = $searchService;
+        $this->container = $container;
+        $this->configResolver = $configResolver;
+
+        return $this;
+    }
+
+    public function redirectToDefinedLocation(GetResponseEvent $event)
+    {
+        if ( $event->getRequestType() == HttpKernelInterface::MASTER_REQUEST ) {
+            $request = $event->getRequest();
+            $locationId = $request->attributes->get('locationId');
+            $requestAttributes = $request->attributes;
+            $siteaccess = $requestAttributes->get('siteaccess')->name;
+
+            if ($siteaccess != 'admin' && $locationId != NULL) {
+                $currentLocation = $this->locationService->loadLocation( $locationId );
+                $currentContent = $this->contentService->loadContentByContentInfo($currentLocation->contentInfo);
+
+                if ( array_key_exists( 'redirect_to' , $currentContent->fields ) && ($currentContent->getFieldValue('redirect_to')->destinationContentId != NULL) && ($currentContent->getFieldValue('redirect_to')->destinationContentId != '')) {
+                    $redirectContent = $this->contentService->loadContent($currentContent->getFieldValue('redirect_to_child'));
+                    $this->redirectByContentInfo($redirectContent->contentInfo, $event);
+                }
+                elseif ( array_key_exists( 'redirect_to_child' , $currentContent->fields ) && $currentContent->getFieldValue('redirect_to_child') == '1') {
+                    $this->redirectToFirstChild($event);
+                }
+            }
+        }
+    }
+
+    public function redirectToFirstChild(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        $locationId = $request->attributes->get('locationId');
+        $location = $this->locationService->loadLocation($locationId);
+        $parentLocation = $this->locationService->loadLocation($location->parentLocationId);
+
+        $redirectCandidates = $this->getRedirectToChildrenCandidates($location->contentInfo);
+
+        $criteria = [
+            new ContentTypeIdentifier($redirectCandidates),
+            new Visibility(Visibility::VISIBLE),
+            new ParentLocationId($location->id)
+        ];
+
+        $query = new LocationQuery();
+        $query->query = new LogicalAnd($criteria);
+        $query->sortClauses = $parentLocation->getSortClauses();
+
+        $blogItemsResult = $this->searchService->findLocations($query);
+
+        if(count($blogItemsResult->searchHits) > 0) {
+            $this->redirectByContentInfo($blogItemsResult->searchHits[0]->valueObject->contentInfo, $event);
```

Comment:
Create method to redirect to location instead of content because `redirectByContentInfo` will redirect to main location